### PR TITLE
Fix: update method in unskript_ctl_db

### PR DESCRIPTION
**Pull request by shloka-bhalgat-unskript**
## Description
1.  Need to handle None type in the convert_memory_to_bytes function
```
k8s:Get K8s services exceeding memory utilization convert_memory_to_bytes function
    Failed Objects:
     '@beartyped check_4.convert_memory_to_bytes() parameter memory_value="None" violates
      type hint <class ''str''>, as "None" not instance of str.'
```
2. Updated the update method to accept byte type object instead of assuming it is dict and added type check.

```
  Traceback (most recent call last):
  File "/usr/local/bin/./unskript_ctl_main.py", line 740, in <module>
    main()
  File "/usr/local/bin/./unskript_ctl_main.py", line 719, in main
    uc.run_main(args=args, parser=parser)
  File "/usr/local/bin/./unskript_ctl_main.py", line 137, in run_main
    self.update_audit_trail(collection_name='audit_trail', status_dict_list=status_of_run)
  File "/usr/local/bin/./unskript_ctl_main.py", line 213, in update_audit_trail
    self._db.pss.update(collection_name=collection_name, data=trail_data)
  File "/usr/local/bin/unskript_ctl_database.py", line 111, in update
    old_data.update(data)
    ^^^^^^^^^^^^^^^
AttributeError: 'bytes' object has no attribute 'update'
```

### Testing

Screenshot 2024-02-14 at 5 10 39 PM

#### Before Fix

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/5c5abf76-9c90-474b-8f4a-ddd4efe85112

#### After Fix & Surviving POD Restart (upgrade scenario)

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/63cc5a53-4585-4a65-b456-4fb3cbdb10d8

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: unskript-ctl/unskript_ctl_database.py [hunk 1]
```diff
@@ -108,7 +109,25 @@ def update(self, **kwargs):
...
+                    self.logger.error(f"Error decoding JSON data: {e}")
+                    return False
+                finally:
+                    self.logger.info("DB schema migrated successfully")
+            elif isinstance(stored_data, dict):
+                old_data = stored_data
+            else:
+                self.logger.error("Unsupported data type in database")
+                return False
             old_data.update(data)
```
Comment (amit-chandak-unskript) [inline unskript-ctl/unskript_ctl_database.py:131]: This is still without try/catch??? Can we handle it gracefully

File: unskript-ctl/unskript_ctl_database.py [hunk 2]
```diff
@@ -108,7 +109,25 @@ def update(self, **kwargs):
...
+                    decompressed_data = zlib.decompress(stored_data)
+                    old_data = json.loads(decompressed_data.decode('utf-8'))
+                except zlib.error as e:
+                    self.logger.error(f"Error decompressing data: {e}")
+                    return False
+                except json.JSONDecodeError as e:
+                    self.logger.error(f"Error decoding JSON data: {e}")
+                    return False
+                finally:
+                    self.logger.info("DB schema migrated successfully")
```
Comment (amit-chandak-unskript) [inline unskript-ctl/unskript_ctl_database.py:125]: Is it the right log? Is it actually migrated? you havent put back the new schema to the disk i think?

Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/unskript_ctl_database.py:127]: Done, had missed adding the SCHEMA_VERSION.

File: unskript-ctl/unskript_ctl_database.py [hunk 3]
```diff
@@ -108,10 +109,37 @@ def update(self, **kwargs):
...
+            else:
+                self.logger.error("Unsupported data type in database")
+                return False
+            try:
+                old_data.update(data)
+                root[self.collection_name] = old_data
+                connection.transaction_manager.commit()
+            except Exception as e:
+                self.logger.debug("Exception Occurred when updating ")
+
```
[the remark this answers is not in the document]
Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/unskript_ctl_database.py:142]: Done.

File: unskript-ctl/unskript_ctl_database.py (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/unskript_ctl_database.py]: Yes. Make sense.

**Comment by jayasimha-raghavan-unskript** [comment]
* This PR brings in support for detecting of OLD schema, Adds schema_version to indicate migration to newer schema
* This ensures the survival of POD deletion & creation (upgrade scenario).

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**